test/unit/resources/mock/aws_ecr_repository_policy_mock.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'aws_base_resource_mock'
+
+class AwsEcrPolicyMock < AwsBaseResourceMock
+  def initialize
+    super
+    @policy_required = {}
+    @policy_required[:repository_name] = @aws.any_arn
+
+    @policy = Hash[@policy_required]
+    @policy[:repository_name] =  "testrepopolicy"  # string
+    @policy[:policy_text] = '{
+      "policyText": "{\n  \"Version\" : \"2012-10-17\",\n  \"Statement\" : [{\n    \"Sid\" : \"AllowPull\",\n    \"Effect\" : \"Allow\",\n    \"Principal\" : \"*\",\n    \"Action\" : [ \"ecr:BatchGetImage\", \"ecr:GetDownloadUrlForLayer\" ]\n  }\n ]\n}",
+      "registryId": "012345678910",
+      "repositoryName": "testrepopolicy"
+    }'
+  end
+
+  def stub_data
+    stub_data = []
+
+    policy = { :client => Aws::ECR::Client,
+               :method => :get_repository_policy,
+               :data => { :policy_text =>  @policy[:policy_text],
+                :repository_name  =>  @policy[:repository_name]}}
```

Comment:
I guess you should not be sending this `repository_name` to stub_data, was this test successful locally ?

Reply:
yes it was successful